lib/DxilPIXPasses/DxilAnnotateWithVirtualRegister.cpp

```diff
@@ -147,12 +175,10 @@ bool DxilAnnotateWithVirtualRegister::runOnModule(llvm::Module &M) {
         skipOverLeadingUnprintableCharacters = 2;
       }
       *OSOverride << fn.second.first << " " << fn.second.second << " "
-                  << (fn.first.str().c_str() +
+                  << (fn.first.c_str() +
```

Comment:
Rather than `c_str()` here, if this were still a StringRef, you could use `StringRef::slice`. Keeping the bounds with the string avoids needing to walk to the null character.

Reply:
Switched to a more string-ref-y approach. Much cleaner. Thanks!